### PR TITLE
[13.0][IMP] analytic_tag_dimension: Improve _reflect function according to prevent error if we use x_dimension_xxx field

### PR DESCRIPTION
**Pull request by victoralmau**
Improve `_reflect` function according to prevent error if we use `x_dimension_xxx` field in inherit models from: 
- account.move.line
- account.analytic.line
- account.invoice.report

Please @pedrobaeza and @carlosdauden can you review it?

@Tecnativa TT29728

### CODE REVIEW

File: analytic_tag_dimension/models/analytic_dimension_line.py [hunk 1]
```diff
@@ -10,6 +11,32 @@ class AnalyticDimensionLine(models.AbstractModel):
...
+            "account.move.line",
+            "account.analytic.line",
+            "account.invoice.report",
+        ]
+
+    def _reflect(self):
+        super()._reflect()
+        if (
+            self._inherit != self._name
+            and self._inherit in self.get_model_names()
```
Comment (pedrobaeza) [inline analytic_tag_dimension/models/analytic_dimension_line.py]: Why are you limiting to these models? Any children model should have it, isn't it?

File: analytic_tag_dimension/models/analytic_dimension_line.py [hunk 2]
```diff
@@ -10,6 +11,32 @@ class AnalyticDimensionLine(models.AbstractModel):
...
+            and self.env.context.get("check_fields", True)
+        ):
+            fields_inherit = self.env["ir.model.fields"].search(
+                [
+                    ("model_id.model", "=", self._inherit),
+                    ("name", "not in", self.fields_get_keys()),
+                ]
+            )
+            if fields_inherit:
+                model = self.env["ir.model"].search([("model", "=", self._name)])
```
Comment (pedrobaeza) [inline analytic_tag_dimension/models/analytic_dimension_line.py]: Why do you need to search again the model if you already have it on any of the fields_inherit records?

File: analytic_tag_dimension/models/analytic_dimension_line.py [hunk 3]
```diff
@@ -10,6 +11,32 @@ class AnalyticDimensionLine(models.AbstractModel):
...
+            for field in fields_inherit:
+                field.with_context(check_fields=False).copy({"model_id": model.id})
```
Comment (pedrobaeza) [inline analytic_tag_dimension/models/analytic_dimension_line.py]: What this exactly does? In the ORM, this is creating a copy of the ir.model.fields record, so not sure this is correct.

Comment (victoralmau) [reply analytic_tag_dimension/models/analytic_dimension_line.py]: It's necessary and create `ir.model.field` records lost.
When dimension record is created, auto-create "custom fields" in some models according to: https://github.com/OCA/account-analytic/blob/13.0/analytic_tag_dimension/models/account_analytic_dimension.py#L38-L59 but these process have side-efects but now work fine.

Example:
- `account.move.line` (inherit from `analytic.dimension.line`) > Before, fields are auto-created in these model.
- `custom.wizard.model` (inherit from `account.move.line`) > Now, fields are auto-created in these model too.

File: analytic_tag_dimension/models/analytic_dimension_line.py (hunk 2 above)
Comment (victoralmau) [reply analytic_tag_dimension/models/analytic_dimension_line.py]: https://github.com/OCA/account-analytic/pull/379#discussion_r634997336

Comment (pedrobaeza) [reply analytic_tag_dimension/models/analytic_dimension_line.py]: I still don't understand how copying a field is the solution. This duplicates the record.

**Review by pedrobaeza — APPROVED** [review]
Checked internally and see the steps to reproduce the problem, that are now in commit message.

**Comment by victoralmau** [comment]
I have added some little changes. 
Now it's ready to merge.

**Pull request merged**